File: 21. Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode *head = new ListNode(0);
        ListNode *prev = head;
        while (l1 != NULL && l2 != NULL) {
            if (l1->val < l2->val) {
                prev->next = l1;
                prev = l1;
                l1 = l1->next;
            } else {
                prev->next = l2;
                prev = l2;
                l2 = l2->next;
            }
        }
        while (l1 != NULL){
            prev->next = l1;
            prev = l1;
            l1 = l1->next;
        }
        while(l2 != NULL){
            prev->next = l2;
            prev = l2;
            l2 = l2->next;
        }
        return head->next;
    }
};

File: 21. Merge Two Sorted Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        dummy_head = ListNode(0)
        prev = dummy_head
        while l1 != None or l2 != None:
            if l1 != None and l2 != None:
                if l1.val < l2.val:                    
                    l3 = ListNode(l1.val)
                    l1 = l1.next
                else:
                    l3 = ListNode(l2.val)
                    l2 = l2.next
            elif l1 != None:
                l3 = ListNode(l1.val)
                l1 = l1.next
            else:
                l3 = ListNode(l2.val)
                l2 = l2.next
            prev.next = l3
            prev = prev.next
        return dummy_head.next